dll/win32/shell32/COpenWithMenu.cpp

```diff
@@ -231,7 +237,17 @@ HICON COpenWithList::GetIcon(SApp *pApp)
         WCHAR wszPath[MAX_PATH];
 
         GetPathFromCmd(wszPath, pApp->wszCmd);
-        ExtractIconExW(wszPath, 0, NULL, &pApp->hIcon, 1);
+        if (!ExtractIconExW(wszPath, 0, NULL, &pApp->hIcon, 1))
+        {
+            SHFILEINFO fi;
+            /* FIXME: Ideally we should include SHGFI_USEFILEATTRIBUTES because we already
+            ** know the file has no icons but SHGetFileInfo is broken in that case (CORE-19122).
+            ** Without SHGFI_USEFILEATTRIBUTES we needlessly hit the disk again but it will
+            ** return the correct default .exe icon.
```

Comment:
Rather uncommon way to comment a block. I would recommend to transfom all "**" to " *" like we do anywhere else.